python/adios2/stream.py

```diff
@@ -270,15 +269,22 @@ def _(self, name, content, shape=[], start=[], count=[], operations=None):
         variable = self._io.inquire_variable(name)
 
         if not variable:
-            # Sequences variables
+            # Sequence variables
             if isinstance(content, np.ndarray):
                 variable = self._io.define_variable(name, content, shape, start, count)
             elif isinstance(content, list):
-                content_np = np.array(content)
-                variable = self._io.define_variable(name, content_np, shape, start, count)
-            # Scalars variables
-            elif isinstance(content, str) or not hasattr(content, "__len__"):
-                variable = self.define_variable(name)
+                content = np.array(content)
+                if shape == []:
+                    shape = [content.size]
+                    count = shape
+                    start = [0]
+                variable = self._io.define_variable(name, content, shape, start, count)
```

Comment:
This will break local variable/arrays since they always use `shape=[]`.

Reply:
Indeed, this broke it but had to add proper parallel testing for local variables to see this. Fixed.